### ISSUE TITLE
CI: test-cdk failed in validate_dashboard for 1.7/stable and 1.8/stable

### ISSUE THREAD

**Issue by kwmonroe** [juju-solutions/bundle-canonical-kubernetes#529]
Both cdk-canal and k8s-core failed their test-cdk jobs because the dashboard returns a 503.

Example from 1.7/stable cdk-canal:
https://ci.kubernetes.juju.solutions/job/test-cdk/452/testReport/junit/(root)/test_cdk/test_deploy_canonical_kubernetes_canal_/

Example from 1.8/stable k8s-core:
https://ci.kubernetes.juju.solutions/job/test-cdk/458/testReport/junit/(root)/test_cdk/test_deploy_kubernetes_core_/

For reference, here are the charm revs from k8s-master/0 during one of the 1.8 failures:
```
# juju status
Model                  Controller      Cloud/Region   Version  SLA
cdk-build-upgrade-565  jenkins-ci-aws  aws/us-east-1  2.3.5    unsupported

App                    Version      Status  Scale  Charm                  Store       Rev  OS      Notes
canal                  0.9.1/2.5.1  active      4  canal                  jujucharms   28  ubuntu
easyrsa                3.0.1        active      1  easyrsa                jujucharms   38  ubuntu
etcd                   3.2.9        active      3  etcd                   jujucharms   76  ubuntu
kubeapi-load-balancer  1.10.3       active      1  kubeapi-load-balancer  jujucharms   56  ubuntu  exposed
kubernetes-e2e         1.8.10       active      1  kubernetes-e2e         jujucharms   59  ubuntu
kubernetes-master      1.8.10       active      1  kubernetes-master      jujucharms   99  ubuntu
kubernetes-worker      1.8.10       active      3  kubernetes-worker      jujucharms  111  ubuntu  exposed
...
```

And snap revs:
```
ubuntu@ip-172-31-61-70:~$ snap list
Name                     Version    Rev   Developer  Notes
cdk-addons               1.8.10     369   canonical  -
core                     16-2.31.2  4206  canonical  core
kube-apiserver           1.8.10     374   canonical  -
kube-controller-manager  1.8.10     365   canonical  -
kube-scheduler           1.8.10     374   canonical  -
kubectl                  1.8.10     374   canonical  classic
```

**Comment by kwmonroe** [comment]
Tried rebuilding cdk-addons to pick up the 1.8.1 dashboard, but this is still failing in 1.8/edge:

https://ci.kubernetes.juju.solutions/job/test-cdk/484/testReport/(root)/test_cdk/test_deploy_kubernetes_core_/

Console shows the latest cdk-addon snap:
```
cdk-addons               1.8.10     380   canonical
```

While the test was running, i tried to hit the url with a browser and got the following:
```
{
  "kind": "Status",
  "apiVersion": "v1",
  "metadata": {
    
  },
  "status": "Failure",
  "message": "no endpoints available for service \"https:kubernetes-dashboard:\"",
  "reason": "ServiceUnavailable",
  "code": 503
}
```

**Comment by kwmonroe** [comment]
Still failing for me in 1.8/stable:

https://ci.kubernetes.juju.solutions/job/test-cdk/668/testReport/junit/(root)/test_cdk/test_deploy_canonical_kubernetes_canal_/
```
            resp = await asyncify(requests.get)(url, auth=auth, verify=False)
            if resp.status_code == 200 and "Dashboard" in resp.text:
                break
            await asyncio.sleep(5)
    
>       assert resp.status_code == 200
E       assert 503 == 200
E        +  where 503 = <Response [503]>.status_code
```

Confirmed this happens with a manual deploy outside of CI too.  It seems the fix is to use the old dashboard url for 1.8:
```python
>>> url = '%s/api/v1/namespaces/kube-system/services/kubernetes-dashboard:/proxy/#!/login' % 'https://18.204.11.254:443'
>>> url
'https://18.204.11.254:443/api/v1/namespaces/kube-system/services/kubernetes-dashboard:/proxy/#!/login'
>>> requests.get(url, auth=auth, verify=False)
<Response [200]>

>>> url = '%s/api/v1/namespaces/kube-system/services/https:kubernetes-dashboard:/proxy/#!/login' % 'https://18.204.11.254:443'
>>> url
'https://18.204.11.254:443/api/v1/namespaces/kube-system/services/https:kubernetes-dashboard:/proxy/#!/login'
>>> requests.get(url, auth=auth, verify=False)
<Response [503]>
```

To do that, I think we just need to adjust the condition to use the old url when < (1, 9) instead of < (1, 8):

https://github.com/juju-solutions/kubernetes-jenkins/blob/master/integration-tests/validation.py#L189

**Comment by kwmonroe** [comment]
@Cynerva noted that our 1.8/stable version of cdk-addons has not been updated to pick up the new 1.8.3 dashboard.  I'll retry once that happens.

**Comment by kwmonroe** [comment]
This is indeed fixed for 1.8 with the latest stable version of cdk-addons.
